lib/generators.nix

```
@@ -597,7 +604,10 @@ in rec {
       (expr "\t${ind}" value)
     ]) x));
 
-  in ''<?xml version="1.0" encoding="UTF-8"?>
+  in
+  # TODO: As discussed in #356502, we should do the actual deprecation in a future release cycle.
+  lib.warnIf (!escape) "Using `lib.generators.toPlist` without escape = true; is deprecated"
```

If you use backticks, use them consistently :-)

I also think that the semi-colon looks bad.

```suggestion
  lib.warnIf (!escape) "Using `lib.generators.toPlist` without `escape = true` is deprecated"
```

Should the warning say "deprecated since XX.YY`? Not sure if we mention release names in `lib/`.

---

Good call; that message was pretty badly formatted.

I was intentionally vague about versioning because I didn't know which XX.YY this PR would become part of or which release the actual removal would happen in.